cmd/kube-controller-manager/app/core.go

```diff
@@ -605,9 +605,11 @@ func setNodeCIDRMaskSizes(cfg nodeipamconfig.NodeIPAMControllerConfiguration) (i
 // then it will return default IPv4 and IPv6 cidr mask sizes.
 func setNodeCIDRMaskSizesDualStack(cfg nodeipamconfig.NodeIPAMControllerConfiguration) (int, int, error) {
 	ipv4Mask, ipv6Mask := defaultNodeMaskCIDRIPv4, defaultNodeMaskCIDRIPv6
-	// NodeCIDRMaskSize can be used only for single stack clusters
-	if cfg.NodeCIDRMaskSize != 0 {
-		return ipv4Mask, ipv6Mask, errors.New("usage of --node-cidr-mask-size is not allowed with dual-stack clusters")
+	// compatible with existing single-stack clusters when dual-stack feature gate become enabled by default
+	if cfg.NodeCIDRMaskSize != 0 && cfg.NodeCIDRMaskSizeIPv4 == 0 && cfg.NodeCIDRMaskSizeIPv6 == 0 {
+		klog.Warning("--node-cidr-mask-size is specified when dual-stack feature gate enabled, please use --node-cidr-mask-size-ipv4 and --node-cidr-mask-size-ipv6 to set IPv4 and IPv6 cidr mask respectively")
+		ipv4Mask, ipv6Mask = int(cfg.NodeCIDRMaskSize), int(cfg.NodeCIDRMaskSize)
```

Comment:
do you think this is a good idea? ipv6 masks can be 128 bits and ipv4 32, 
If I set a /64 mask, it won't work for ipv4.
If I set a /12 mask it will force the node mask to be /28, that's a huge overlap with other networks, I don't think is feasible having an IPv6 /12 in production

Reply:
Emm... You're right. More validations are needed here.
How about following the clusterCIDR kind? 
If it is IPv4, only set ipv4Mask, leave ipv6Mask as default.
If it is IPv6, only set ipv6Mask, leave ipv4Mask as default.